src/ucs/sys/sys.c

```diff
@@ -171,7 +171,7 @@ static uint64_t __sumup_host_name(unsigned prime_index)
     p = ucs_get_host_name();
     while (*p != '\0') {
         n = 0;
-        strncpy((char*)&n, p, sizeof(n));
+        memcpy(&n, p, strnlen(p, sizeof(n)));
```

Comment:
shy not ucs_safe_strncpy()?

Reply:
because it set last symbol to '\0' which may cause collision (for example in case if host names are differ in 4-th symbol only)